CLI: segmentation violation when using update/upgrade without ID

### What happened?

`grafana-cli plugins update` and `upgrade` crash.

```
grafana-cli plugins update
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x9cfc116]

goroutine 1 [running]:
github.com/grafana/grafana/pkg/cmd/grafana-cli/commands.shouldUpgrade({{{0xc001397a80, 0x1c}, {0xc0029cb6e0, 0xa}, {0xc0029cb6d0, 0x8}, {0x0, 0x0, 0x0}, {{{...}, ...}, ...}, ...}, ...}, ...)
	github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/upgrade_all_command.go:26 +0x1d6
github.com/grafana/grafana/pkg/cmd/grafana-cli/commands.upgradeCommand({0x22527e0, 0xc00268ab48})
	github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/upgrade_command.go:29 +0x257
github.com/grafana/grafana/pkg/cmd/grafana-cli/commands.init.runPluginCommand.func4(0xc002870000)
	github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/commands.go:78 +0x5c
github.com/urfave/cli/v2.(*Command).Run(0xa4cce80, 0xc002870000, {0xc00286a790, 0x1, 0x1})
	github.com/urfave/cli/v2@v2.25.0/command.go:273 +0x998
github.com/urfave/cli/v2.(*Command).Run(0xa4ce5e0, 0xc002823ec0, {0xc00286e4e0, 0x2, 0x2})
	github.com/urfave/cli/v2@v2.25.0/command.go:266 +0xbe5
github.com/urfave/cli/v2.(*Command).Run(0xc00269b600, 0xc002823b80, {0xc00284e8c0, 0x7, 0x7})
	github.com/urfave/cli/v2@v2.25.0/command.go:266 +0xbe5
github.com/urfave/cli/v2.(*Command).Run(0xc002860000, 0xc002823a40, {0xc00012e500, 0x8, 0x8})
	github.com/urfave/cli/v2@v2.25.0/command.go:266 +0xbe5
github.com/urfave/cli/v2.(*App).RunContext(0xc00285e000, {0x2216700?, 0xa55d860}, {0xc00012e500, 0x8, 0x8})
	github.com/urfave/cli/v2@v2.25.0/app.go:332 +0x5b7
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.25.0/app.go:309
main.main()
	github.com/grafana/grafana/pkg/cmd/grafana/main.go:54 +0xb2b
```

### What did you expect to happen?

No segmentation fault, and instead it should mention that an ID needs to be specified.

### Did this work before?

Yes.

### How do we reproduce it?

1. Run `grafana-cli plugins` with either `update` or `upgrade`

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

```
$ grafana-cli --version
grafana version 10.4.1
```

```
$ grafana-server --version
Version 10.4.1 (commit: d3ce857c0eb86f571ffa993a9cd8493b6f47b630, branch: HEAD)
```

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

### Datasource(s)?

_No response_

--- a/pkg/cmd/grafana-cli/commands/upgrade_command.go
+++ b/pkg/cmd/grafana-cli/commands/upgrade_command.go
@@ -15,10 +15,9 @@ import (
 func upgradeCommand(c utils.CommandLine) error {
 	ctx := context.Background()
 	pluginsDir := c.PluginDirectory()
-
 	pluginID := c.Args().First()
 	if pluginID == "" {
-		return errors.New("missing plugin parameter")
+		return errors.New("please specify plugin to update")
 	}
 
 	localPlugin, err := services.GetLocalPlugin(pluginsDir, pluginID)

--- a/pkg/cmd/grafana-cli/commands/upgrade_command.go
+++ b/pkg/cmd/grafana-cli/commands/upgrade_command.go
@@ -2,6 +2,7 @@ package commands
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"github.com/fatih/color"
@@ -14,7 +15,11 @@ import (
 func upgradeCommand(c utils.CommandLine) error {
 	ctx := context.Background()
 	pluginsDir := c.PluginDirectory()
+
 	pluginID := c.Args().First()
+	if pluginID == "" {
+		return errors.New("missing plugin parameter")
+	}
 
 	localPlugin, err := services.GetLocalPlugin(pluginsDir, pluginID)
 	if err != nil {

